src/librustc_codegen_llvm/back/link.rs

```diff
@@ -68,36 +68,21 @@ pub fn get_linker(sess: &Session) -> (PathBuf, Command) {
     // This worked historically but is needed manually since #42436 (regression
     // was tagged as #42791) and some more info can be found on #44443 for
     // emscripten itself.
-    let cmd = |linker: &Path| {
+    let mut cmd = (|| {
         if let Some(linker) = linker.to_str() {
             if cfg!(windows) && linker.ends_with(".bat") {
                 return Command::bat_script(linker)
             }
         }
-        match sess.linker_flavor() {
+        match flavor {
             LinkerFlavor::Lld(f) => Command::lld(linker, f),
             _ => Command::new(linker),
 
         }
-    };
+    })();
```

Comment:
You can probably remove the intermediate closure here, I don't think it's serving much purpose any more